portal-api/src/cron-scheduler-jobs/helpers/utilisation-report-helpers.ts

```diff
@@ -0,0 +1,203 @@
+import { format, subMonths } from 'date-fns';
+import { MonthAndYear, OneIndexedMonth, PaymentOfficerTeam, ReportPeriod, getCurrentReportPeriodForBankSchedule, getOneIndexedMonth } from '@ukef/dtfs2-common';
+import externalApi from '../../external-api/api';
+import api from '../../v1/api';
+import { getBusinessDayOfMonth } from '../../utils/date';
+import { isValidEmail } from '../../utils/string';
+import { BANK_HOLIDAY_REGION } from '../../constants/bank-holiday-region';
+import { BankResponse } from '../../v1/api-response-types';
+
+export type SendEmailCallback = (emailAddress: string, recipient: string, formattedReportPeriod: string) => Promise<void>;
```

Comment:
I noticed that the old parameter type for this callback used to be `({ emailAddress, recipient, formattedReportPeriod }: { emailAddress: string; recipient: string; formattedReportPeriod: string })` - just wondering what the reason for this change was :)

Reply:
My assumption was it was written that way due to it being javascript to make it clearer what the parameters were, since we did have a comment detailing the type but it wasn't attached to the usages very clearly so hard to see when doing dev. Since now it is in typescript it was easier to determine parameters and their types and I think it is more readable to not bundle the parameters into an object only to immediately deconstruct them.

Reply:
I think the state machine is quite different, we have a lot of callbacks that we want to nominally have the same shape, either they do or don't take a report, and then they have a bunch of other stuff that differs per event which we wrap up in a payload, so here it makes sense to have an object so that we can support that nicely. I think if we had a few more parameters here then we might want to group them up in a semantic way into one or more objects to keep the code nice and readable, like we do with the state machine